File: sandbox/aiml/betty/src/main/scala/bodl/sandbox/aiml/betty/internal/ExampleActivator.scala
package bodl.sandbox.aiml.betty.internal;

import java.util.Properties;

import org.osgi.framework.BundleActivator;
import org.osgi.framework.BundleContext;

import bodl.sandbox.aiml.betty.ExampleService;

class ExampleActivator
    extends BundleActivator
{
    /**
     * Called whenever the OSGi framework starts our bundle
     */
    override def start(bc:BundleContext) = {
        println( "STARTING bodl.sandbox.helloscala" )

        val props = new Properties()
        // add specific service properties here...

        println( "REGISTER bodl.sandbox.helloscala.ExampleService" )

        // Register our example service implementation in the OSGi service registry
        val impl = new ExampleServiceImpl()
        bc.registerService( classOf[ExampleService].getName(), impl, props )

        val hello = impl.scramble("Hello World")
    }

    /**
     * Called whenever the OSGi framework stops our bundle
     */
    override def stop(bc:BundleContext) =
    {
        println( "STOPPING bodl.sandbox.helloscala" )

        // no need to unregister our service - the OSGi framework handles it for us
    }
}

File: sandbox/jersey/jacksonjsonprovider/src/main/java/bodl/sandbox/jersey/jacksonjsonprovider/ExampleActivator.java
package bodl.sandbox.jersey.jacksonjsonprovider;

import java.util.Dictionary;
import java.util.HashSet;
import java.util.Hashtable;
import java.util.Set;

import javax.servlet.Servlet;
import javax.ws.rs.core.Application;

import org.osgi.framework.BundleActivator;
import org.osgi.framework.BundleContext;

import com.sun.jersey.spi.container.servlet.ServletContainer;

/**
 * Extension of the default OSGi bundle activator
 */
public final class ExampleActivator
    implements BundleActivator
{
    /**
     * Called whenever the OSGi framework starts our bundle
     */
    public void start( BundleContext bc )
        throws Exception
    {
        System.out.println( "STARTING bodl.sandbox.jersey.jacksonjsonprovider" );

        Dictionary props = new Hashtable();
        props.put( "alias", "/sandbox/jersey/jacksonjsonprovider" );
        props.put("servlet-name", "bodl-sandbox jersey jackson");

        System.out.println( "REGISTER bodl.sandbox.jersey.jacksonjsonprovider.ServletContainer" );

		ServletContainer sc = new com.sun.jersey.spi.container.servlet.ServletContainer(new Application() {
			public Set<Class<?>> getClasses() {
				Set<Class<?>> s = new HashSet<Class<?>>();
				s.add(EmptyArrayResource.class);
				s.add(NonJAXBBeanResource.class);
				s.add(org.codehaus.jackson.jaxrs.JacksonJsonProvider.class);
				return s;
			}
		});
        // Register our example service implementation in the OSGi service registry
        bc.registerService( Servlet.class.getName(), sc, props );
    }

    /**
     * Called whenever the OSGi framework stops our bundle
     */
    public void stop( BundleContext bc )
        throws Exception
    {
        System.out.println( "STOPPING bodl.sandbox.paxwhite" );

        // no need to unregister our service - the OSGi framework handles it for us
    }
}

